src/slider/internal.tsx

```diff
@@ -0,0 +1,188 @@
+import Tooltip from '../internal/components/tooltip/index.js';
+import SliderLabels from './slider-labels.js';
+import { getPercent, getStepArray, findLowerAndHigherValues, valuesAreValid, THUMB_SIZE } from './utils.js';
+import customCssProps from '../internal/generated/custom-css-properties/index.js';
+import { warnOnce } from '@cloudscape-design/component-toolkit/internal';
+
+export default function InternalSlider({
+  value,
+  min,
+  max,
+  onChange,
+  step,
+  disabled,
+  ariaLabel,
+  referenceValues,
+  tickMarks,
+  hideFillLine,
+  valueFormatter,
+  ...rest
+}: SliderProps) {
+  const { __internalRootRef } = useBaseComponent('Slider');
+  const baseProps = getBaseProps(rest);
+
+  const handleRef = useRef<HTMLDivElement>(null);
+  const [showTooltip, setShowTooltip] = useState(false);
+  const [isActive, setIsActive] = useState(false);
+  const labelsId = useUniqueId('labels');
+  const hasValue = value !== undefined;
+  const formFieldContext = useFormFieldContext(rest);
+  const { ariaLabelledby, ariaDescribedby, controlId, invalid } = formFieldContext ?? rest;
+
+  const percent = hasValue && getPercent(Math.max(Math.min(value, max), min), [min, max]);
+
+  // Because absolute positioning does not center elements, we need to add an offset
+  // so that the tooltip is centered over the value.
+  // At the min and max , the offset is half the size of the thumb to
+  // keep it from bleeding outside of the bounds of the slider. Otherwise, the
+  // offset is the size of the thumb.
+  const tooltipPositionAdjust =
+    hasValue && (value <= min ? THUMB_SIZE - THUMB_SIZE / 2 : value >= max ? THUMB_SIZE + THUMB_SIZE / 2 : THUMB_SIZE);
+
+  if (referenceValues && valuesAreValid(referenceValues) === false) {
+    warnOnce('Slider', 'All reference values must be integers. Non-integer values will not be displayed.');
+  }
+
+  if (min >= max) {
+    warnOnce('Slider', 'The min value cannot be greater than the max value.');
+  }
+
+  if (step && step > max - min) {
+    warnOnce('Slider', 'The step value cannot be greater than the difference between the min and max.');
+  }
+
+  const tooltip = hasValue && <Tooltip value={valueFormatter ? valueFormatter(value) : value} trackRef={handleRef} />;
+
+  const getAriaValueText = () => {
+    if (value !== undefined && valueFormatter && valueFormatter(value)) {
+      return valueFormatter(value);
+    }
+
+    if (value && referenceValues && valueFormatter && !valueFormatter(value)) {
+      const valueArray = [min, ...referenceValues, value, max];
+      const prevAndNext = findLowerAndHigherValues(valueArray, value);
+      const prev = prevAndNext.lower && valueFormatter(prevAndNext.lower);
+      const next = prevAndNext.higher && valueFormatter(prevAndNext.higher);
+
+      if (prev === undefined) {
+        return `${value}, between ${valueFormatter(min)} and ${next}`;
+      }
+      if (next === undefined) {
+        return `${value}, between ${prev} and ${valueFormatter(max)}`;
```

Comment:
The hardcoded labels need to be replaced with i18n and tested with I18nProvider.

Reply:
Yes, I have this on my to-do list. Thanks for the reminder.